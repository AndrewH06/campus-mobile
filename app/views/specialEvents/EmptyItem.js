import React from 'react';
import {
	View,
	StyleSheet,
} from 'react-native';

import {
	COLOR_LGREY,
} from '../../styles/ColorConstants';

const EmptyItem = () => (
	<View style={styles.emptyRow} />
);

const styles = StyleSheet.create({
	emptyRow: { width: 75, flexDirection: 'row',  backgroundColor: COLOR_LGREY },
});

export default EmptyItem;
