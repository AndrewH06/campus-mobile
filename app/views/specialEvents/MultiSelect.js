import React, { Component } from 'react';
import {
	ListView,
	StyleSheet,
	View,
	Text,
} from 'react-native';
import { Actions } from 'react-native-router-flux';
import Touchable from '../common/Touchable';
import CircleCheckBox from '../common/CircleCheckBox';
import {
	COLOR_LGREY,
	COLOR_DGREY,
	COLOR_WHITE,
	COLOR_PRIMARY,
} from '../../styles/ColorConstants';
import { WINDOW_WIDTH } from '../../styles/LayoutConstants';

/**
 * @param {String[]} items - items to be selected
 * @param {String[]} selected - items already selected
 * @param {Object} themes - coloring for items
 * @param {Function} onSelect
 */
class MultiSelect extends Component {
	// Returns true if item is being selected
	// false if being de-selected
	onSelect = (item) => {
		const currSelected = this.props.selected.slice();
		const itemIndex = currSelected.indexOf(item);

		if (itemIndex > -1) {
			currSelected.splice(itemIndex, 1);
			this.props.onSelect(currSelected);
			return true;
		} else {
			currSelected.push(item);
			this.props.onSelect(currSelected);
			return false;
		}
	}

	onClear = () => {
		this.props.onSelect([]);
	}

	render() {
		const { items, themes, selected } = this.props;
		const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
		const dataSource = ds.cloneWithRows(items);

		return (
			<View style={styles.container}>
				<ListView
					dataSource={dataSource}
					renderRow={(rowData) => (
						<MultiSelectItem
							data={rowData}
							selected={selected.includes(rowData)}
							onSelect={this.onSelect}
							color={(themes && themes[rowData] !== '') ? (themes[rowData]) : (COLOR_DGREY)}
						/>
					)}
				/>

				<Touchable
					onPress={() => this.props.applyFilters()}
					style={styles.applyButton}
				>
					<Text style={styles.applyText}>Apply</Text>
				</Touchable>
			</View>
		);
	}
}

const MultiSelectItem = ({ data, selected, onSelect, color }) => (
	<Touchable
		onPress={() => onSelect(data)}
		style={styles.itemRow}
	>
		<CircleCheckBox
			checked={selected}
			label={data}
			onToggle={() => onSelect(data)}
			outerColor={color}
			innerColor={color}
		/>
	</Touchable>
);

const styles = StyleSheet.create({
	container: { backgroundColor: 'white', paddingBottom: 60, borderWidth: 1 },
	itemRow: { flexDirection: 'row', height: 50, alignItems: 'center', backgroundColor: 'white', borderBottomWidth: 1, borderBottomColor: COLOR_LGREY, paddingHorizontal: 12 },
	applyButton: { position: 'absolute', width: Math.round(WINDOW_WIDTH * .9), height: 40, bottom: 10, left: Math.round(WINDOW_WIDTH * .05), justifyContent: 'center', alignItems: 'center', backgroundColor: COLOR_PRIMARY, borderRadius: 3, padding: 10 },
	applyText: { fontSize: 16, color: COLOR_WHITE },
});

export default MultiSelect;
