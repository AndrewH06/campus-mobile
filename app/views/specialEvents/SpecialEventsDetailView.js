import React from 'react';
import {
	View,
	Text,
	ScrollView,
	StyleSheet,
} from 'react-native';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/Ionicons';
import moment from 'moment';

import Touchable from '../common/Touchable';
import css from '../../styles/css';
import logger from '../../util/logger';
import { gotoNavigationApp, getHumanizedDuration, platformIOS } from '../../util/general';
import {
	COLOR_DGREY,
	COLOR_MGREY,
	COLOR_PRIMARY,
	COLOR_BLACK,
	COLOR_YELLOW,
} from '../../styles/ColorConstants';
import {
	WINDOW_WIDTH
} from '../../styles/LayoutConstants';

const SpecialEventsDetailView = ({ data, saved, add, remove }) => {
	logger.ga('View Loaded: SpecialEvents Detail: ' + data['talk-title']);

	return (
		<View style={[css.main_container, css.whitebg]}>
			<ScrollView>
				<View style={styles.detailContainer}>
					<View style={styles.starButton}>
						<Touchable
							onPress={
								() => (isSaved(saved, data.id) ? (remove(data.id)) : (add(data.id)))
							}
						>
							<View style={styles.starButtonInner}>
								<Icon
									name={'ios-star-outline'}
									size={32}
									style={styles.starOuterIcon}
								/>
								{ isSaved(saved, data.id)  ? (
									<Icon
										name={'ios-star'}
										size={26}
										style={styles.starInnerIcon}
									/>
								) : null }
							</View>
						</Touchable>
					</View>

					<View style={styles.labelView}>
						{ data.label ? (
							<Text style={[styles.labelText, { color: data['label-theme'] ? data['label-theme'] : COLOR_BLACK }]}>{data.label}</Text>
						) : null }
						{ data['talk-type'] === 'Keynote' ? (
							<Text style={styles.labelText}>{data['talk-type']}</Text>
						) : null }
						{ data.label || data['talk-type'] === 'Keynote' ? (
							<Text style={styles.labelText}> - </Text>
						) : null }
						<Text style={styles.labelText}>{getHumanizedDuration(data['start-time'], data['end-time'])}</Text>
					</View>

					<Text style={styles.sessionName}>
						{data['talk-title']}
					</Text>
					<Text style={styles.sessionInfo}>
						{data.location} - {moment(Number(data['start-time'])).format('MMM Do YYYY, h:mm a')}
					</Text>

					<Text style={styles.sessionDesc}>
						{data['full-description']}
					</Text>

					{(data.directions && data.directions.latitude && data.directions.longitude) ? (
						<Touchable
							underlayColor={'rgba(200,200,200,.1)'}
							onPress={() => gotoNavigationApp(data.directions.latitude, data.directions.longitude)}
						>
							<View style={styles.sed_dir}>
								<Text style={styles.sed_dir_label}>Directions</Text>
								<Icon name="md-walk" size={32} style={styles.sed_dir_icon} />
							</View>
						</Touchable>
					) : null }


					{data.speakers ? (
						<View>
							<Text style={styles.hostedBy}>Hosted By</Text>
							{data.speakers.map((object, i) => (
								<View style={styles.speakerContainer} key={i}>
									<Text style={styles.speakerName}>{object.name}</Text>
									<Text style={styles.speakerPosition}>{object.position}</Text>
									{/*<Text style={styles.speakerSubTalkTitle}>{object['sub-talk-title']}</Text>*/}
								</View>
							))}
						</View>
					) : null }
				</View>
			</ScrollView>
		</View>
	);
};

function isSaved(savedArray, id) {
	if (Array.isArray(savedArray)) {
		for ( let i = 0; i < savedArray.length; ++i) {
			if (savedArray[i] === id) {
				return true;
			}
		}
	}
	return false;
}

const mapStateToProps = (state) => (
	{
		saved: state.specialEvents.saved
	}
);

const ActualSpecialEventsDetailView = connect(
	mapStateToProps
)(SpecialEventsDetailView);

const styles = StyleSheet.create({
	detailContainer: { width: WINDOW_WIDTH, padding: 12 },
	labelView: { flexDirection: 'row', paddingTop: 4 },
	labelText: { fontSize: 13 },
	sessionName: { fontSize: 24, color: COLOR_PRIMARY, paddingTop: 6 },
	sessionInfo: { fontSize: 12, paddingTop: 16  },
	sessionDesc: { lineHeight: 18, fontSize: 14, paddingTop: 16 },
	hostedBy: { fontSize: 10, fontWeight: 'bold', marginTop: 16 },
	speakerContainer: { marginTop: 2 },
	speakerName: { fontSize: 14, fontWeight: 'bold', color: COLOR_PRIMARY, marginTop: 10 },
	speakerPosition: { fontSize: 10, marginTop: 2 },
	speakerSubTalkTitle: { fontSize: 10, marginTop: 2 },
	starButton: { width: 50, position: 'absolute', top: 2, right: -5, zIndex: 10 },
	starButtonInner: { justifyContent: 'flex-start', alignItems: 'center' },
	starOuterIcon: { color: COLOR_DGREY, position: platformIOS() ? 'absolute' : 'relative', zIndex: 10, backgroundColor: 'transparent' },
	starInnerIcon: { color: COLOR_YELLOW, position: 'absolute', zIndex: platformIOS() ? 5 : 15, marginTop: 3 },
	sed_dir: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', borderTopWidth: 1, borderTopColor: COLOR_MGREY, marginTop: 16, paddingVertical: 6 },
	sed_dir_label: { flex: 1, fontSize: 22, color: COLOR_PRIMARY },
	sed_dir_icon: { color: COLOR_PRIMARY, alignSelf: 'flex-end' },
});

export default ActualSpecialEventsDetailView;
