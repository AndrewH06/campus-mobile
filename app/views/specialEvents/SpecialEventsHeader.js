import React from 'react';
import {
	View,
	Text,
	StyleSheet,
} from 'react-native';
import moment from 'moment';

import {
	COLOR_LGREY,
	COLOR_BLACK,
} from '../../styles/ColorConstants';

const SpecialEventsHeader = ({ timestamp }) => (
	<View
		style={styles.header}
	>
		<Text style={styles.headerText}>
			{moment(Number(timestamp)).format('MM/DD hh:mm')}
		</Text>
	</View>
);

const styles = StyleSheet.create({
	header: { justifyContent: 'flex-start', alignItems: 'center', width: 45, backgroundColor: COLOR_LGREY, borderBottomWidth: 1, borderColor: COLOR_LGREY },
	headerText: { textAlign: 'right', alignSelf: 'stretch', color: COLOR_BLACK, fontSize: 12, marginTop: 7 },
});

export default SpecialEventsHeader;
