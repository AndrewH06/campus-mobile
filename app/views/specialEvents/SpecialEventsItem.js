import React from 'react';
import {
	View,
	Text,
	StyleSheet,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Actions } from 'react-native-router-flux';

import Touchable from '../common/Touchable';
import { platformIOS, getHumanizedDuration } from '../../util/general';
import {
	COLOR_DGREY,
	COLOR_MGREY,
	COLOR_LGREY,
	COLOR_BLACK,
	COLOR_YELLOW
} from '../../styles/ColorConstants';

const SpecialEventsItem = ({ specialEventsData, saved, add, remove }) => (
	<View
		style={styles.itemRow}
	>
		<CircleBorder />
		<View style={styles.titleContainer}>
			<Touchable
				onPress={() => Actions.SpecialEventsDetailView({ data: specialEventsData, add, remove })}
			>
				<View>
					{specialEventsData['talk-title'] ? (
						<Text
							style={styles.titleText}
							numberOfLines={2}
						>
							{specialEventsData['talk-title']}
						</Text>
					) : null }

					<View style={styles.labelView}>
						{ specialEventsData.label ? (
							<Text style={[styles.labelText, { color: specialEventsData['label-theme'] ? specialEventsData['label-theme'] : COLOR_BLACK }]}>{specialEventsData.label}</Text>
						) : null }
						{ specialEventsData['talk-type'] === 'Keynote' ? (
							<Text style={styles.labelText}>{specialEventsData['talk-type']}</Text>
						) : null }
						{ specialEventsData.label || specialEventsData['talk-type'] === 'Keynote' ? (
							<Text style={styles.labelText}> - </Text>
						) : null }
						<Text style={styles.labelText}>{getHumanizedDuration(specialEventsData['start-time'], specialEventsData['end-time'])}</Text>
					</View>
				</View>
			</Touchable>
		</View>

		{ (add !== null) ? (
			<Touchable
				style={styles.starButton}
				onPress={
					() => ((saved) ? (remove(specialEventsData.id)) : (add(specialEventsData.id)))
				}
			>
				<View style={styles.starButtonInner}>
					<Icon
						name={'ios-star-outline'}
						size={32}
						style={styles.starOuterIcon}
					/>
					{ saved ? (
						<Icon
							name={'ios-star'}
							size={26}
							style={styles.starInnerIcon}
						/>
					) : null }
				</View>
			</Touchable>
		) : null }
	</View>
);

const CircleBorder = () => (
	<View style={styles.borderContainer}>
		<View style={styles.line} />
		<View style={styles.circle} />
	</View>
);

const styles = StyleSheet.create({
	itemRow: { flexGrow: 1, flexDirection: 'row', backgroundColor: COLOR_LGREY },
	titleContainer: { flex: 1, marginTop: 3 },
	titleText: { alignSelf: 'stretch', fontSize: 17, color: '#000' },
	labelView: { flexDirection: 'row', paddingTop: 4 },
	labelText: { fontSize: 13 },
	starButton: { width: 50 },
	starButtonInner: { justifyContent: 'flex-start', alignItems: 'center' },
	starOuterIcon: { color: COLOR_DGREY, position: platformIOS() ? 'absolute' : 'relative', zIndex: 10, backgroundColor: 'rgba(0,0,0,0)' },
	starInnerIcon: { color: COLOR_YELLOW, position: 'absolute', zIndex: platformIOS() ? 5 : 15, marginTop: 3 },
	borderContainer: { width: 1, alignSelf: 'stretch', marginHorizontal: 10, alignItems: 'flex-start' },
	line: { flexGrow: 1, borderLeftWidth: 1, borderColor: COLOR_MGREY, paddingBottom: 20 },
	circle: { position: 'absolute', top: 11, left: -2.5, height: 6, width: 6, borderRadius: 3, borderWidth: 1, borderColor: COLOR_MGREY, backgroundColor: COLOR_LGREY },
});

export default SpecialEventsItem;
