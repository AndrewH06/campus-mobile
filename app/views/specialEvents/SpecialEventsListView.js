import React from 'react';
import {
	View,
	Text,
	ListView,
	StyleSheet,
} from 'react-native';
import { connect } from 'react-redux';
import moment from 'moment';

import {
	MAX_CARD_WIDTH,
	WINDOW_WIDTH,
	WINDOW_HEIGHT,
	NAVIGATOR_HEIGHT,
	TAB_BAR_HEIGHT,
} from '../../styles/LayoutConstants';
import EmptyItem from './EmptyItem';
import SpecialEventsItem from './SpecialEventsItem';
import SpecialEventsHeader from './SpecialEventsHeader';


const dataSource = new ListView.DataSource({
	rowHasChanged: (r1, r2) => r1 !== r2,
	sectionHeaderHasChanged: (s1, s2) => s1 !== s2
});

const SpecialEventsListView = ({ addSpecialEvents, specialEventsSchedule,
	specialEventsScheduleIds, removeSpecialEvents, saved,disabled, personal, rows,
	scrollEnabled, style, labels, labelItemIds, selectedDay, days, daysItemIds, inCard }) => {

	let scheduleIdArray = [];
	// Use ids from selectedDay
	if (daysItemIds) {
		if (!selectedDay) {
			// Select current day by default
			for (let i = 0; i < days.length; ++i) {
				selectedDay = days[i];
				if (moment(selectedDay).isSameOrAfter(moment(), 'day')) {
					scheduleIdArray = daysItemIds[selectedDay];

					// Filter saved for day
					if (personal && Array.isArray(saved)) {
						scheduleIdArray = scheduleIdArray.filter((item) => saved.includes(item));
					}

					// If displaying for card...continue looking for a day with saved
					if (inCard && scheduleIdArray.length === 0) {
						// continue
					} else {
						break;
					}
				}
			}
		}
		scheduleIdArray = daysItemIds[selectedDay];

		// Filter saved for day
		if (personal && Array.isArray(saved)) {
			scheduleIdArray = scheduleIdArray.filter((item) => saved.includes(item));
		}

		// Apply label filtering
		if (!personal && labels.length > 0) {
			let labelArray = [];
			for (let j = 0; j < labels.length; ++j) {
				const label = labels[j];
				const items = labelItemIds[label];

				labelArray = labelArray.concat(items);
			}
			scheduleIdArray = scheduleIdArray.filter((item) => labelArray.includes(item));
		}
	}
	if (personal && scheduleIdArray.length === 0) {
		return (
			<View style={[style, rows ? styles.card : styles.full]}>
				<Text style={styles.noSavedSessions}>
					Click the star icon next to a session to save it to your list.
				</Text>
			</View>
		);
	} else {
		return (
			<ListView
				style={[style, rows ? styles.card : styles.full]}
				scrollEnabled={scrollEnabled}
				stickySectionHeadersEnabled={false}
				dataSource={
					dataSource.cloneWithRowsAndSections(
						convertToTimeMap(
							specialEventsSchedule,
							adjustData(specialEventsSchedule, scheduleIdArray, saved, personal, rows)
						)
					)
				}
				renderRow={(rowData, sectionID, rowID, highlightRow) => {
					// Don't render first row bc rendered by header
					if (Number(rowID) !== 0) {
						return (
							<View style={styles.rowContainer}>
								<EmptyItem />
								<SpecialEventsItem
									specialEventsData={rowData}
									saved={saved.includes(rowData.id)}
									add={(disabled) ? null : addSpecialEvents}
									remove={removeSpecialEvents}
								/>
							</View>
						);
					} else {
						return null;
					}
				}}
				renderSectionHeader={(sectionData, sectionID) => (
					// Render header along with first row
					<View style={styles.rowContainer}>
						<SpecialEventsHeader
							timestamp={sectionID}
						/>
						<SpecialEventsItem
							specialEventsData={sectionData[0]}
							saved={saved.includes(sectionData[0].id)}
							add={(disabled) ? null : addSpecialEvents}
							remove={removeSpecialEvents}
						/>
					</View>
				)}
			/>
		);
	}
};

/*
	Filters what session ids to use based on personal/saved and/or rows
	@returns Array of session ids
 */
function adjustData(scheduleIdMap, scheduleIdArray, savedArray, personal, rows) {
	// Filter out saved items
	if (!personal || !Array.isArray(savedArray)) {
		const keys = scheduleIdArray;
		if (!rows) {
			return scheduleIdArray;
		} else {
			const trimmed = [];

			for (let i = 0; i < rows; ++i) {
				trimmed.push(keys[i]);
			}
			return trimmed;
		}
	} else {
		let filtered = [];
		// Check if saved item is part of ids to be displayed
		for (let i = 0; i < savedArray.length; ++i) {
			const key = savedArray[i];
			if (scheduleIdArray.includes(key)) {
				filtered.push(key);
			}
		}

		// Displaying for homecard
		// Remove sessions that have passed
		if (rows) {
			const now = Date.now();
			if (filtered.length > rows) {
				const temp = filtered.slice();
				for (let j = 0; j < filtered.length; ++j) {
					const key = filtered[j];
					const itemTime = scheduleIdMap[key]['start-time'];

					if (now > itemTime) {
						const index = temp.indexOf(key);
						temp.splice(index, 1);
					}

					if (temp.length <= rows) {
						break;
					}
				}

				if (temp.length > rows) {
					filtered = temp.slice(0, rows);
				} else {
					filtered = temp;
				}
			}
		}
		return filtered;
	}
}

/*
	@returns Object that maps keys from scheduleArray to Objects in scheduleMap
*/
function convertToTimeMap(scheduleMap, scheduleArray, header = false) {
	const timeMap = {};

	if (Array.isArray(scheduleArray)) {
		scheduleArray.forEach((key) => {
			const session = scheduleMap[key];
			if (!timeMap[session['start-time']]) {
				// Create an entry in the map for the timestamp if it hasn't yet been created
				timeMap[session['start-time']] = [];
			}
			timeMap[session['start-time']].push(session);
		});

		// Remove an item from section so spacing lines up
		if (header) {
			Object.keys(timeMap).forEach((key) => {
				timeMap[key].pop();
			});
		}
	}
	return timeMap;
}

const mapStateToProps = (state) => (
	{
		specialEventsSchedule: state.specialEvents.data.schedule,
		specialEventsScheduleIds: state.specialEvents.data.uids,
		saved: state.specialEvents.saved,
		labelItemIds: state.specialEvents.data['label-items'],
		labels: state.specialEvents.labels,
		days: state.specialEvents.data.dates,
		daysItemIds: state.specialEvents.data['date-items'],
	}
);

const mapDispatchToProps = (dispatch) => (
	{
		addSpecialEvents: (id) => {
			dispatch({ type: 'ADD_SPECIAL_EVENTS', id });
		},
		removeSpecialEvents: (id) => {
			dispatch({ type: 'REMOVE_SPECIAL_EVENTS', id });
		}
	}
);

const ActualSpecialEventsListView = connect(
	mapStateToProps,
	mapDispatchToProps
)(SpecialEventsListView);

const styles = StyleSheet.create({
	rowContainer: { flexDirection: 'row', height: 76 },
	full: { width: WINDOW_WIDTH, height: (WINDOW_HEIGHT - NAVIGATOR_HEIGHT - TAB_BAR_HEIGHT) },
	card: { width: MAX_CARD_WIDTH },
	noSavedSessions: { flexGrow: 1, fontSize: 18, textAlign: 'center', padding: 40, lineHeight: 30 },
});

export default ActualSpecialEventsListView;
