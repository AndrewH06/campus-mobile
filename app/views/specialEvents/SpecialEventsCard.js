import React from 'react';
import {
	View,
	Text,
	StyleSheet,
} from 'react-native';
import { Actions } from 'react-native-router-flux';

import Card from '../card/Card';
import BannerCard from '../common/BannerCard';
import SpecialEventsListView from './SpecialEventsListView';
import Touchable from '../common/Touchable';
import {
	COLOR_DGREY,
	COLOR_MGREY,
	COLOR_PRIMARY,
} from '../../styles/ColorConstants';
import {
	MAX_CARD_WIDTH,
} from '../../styles/LayoutConstants';

export const SpecialEventsCard = ({ specialEvents, saved, hideCard }) => (
	<View>
		{ (Array.isArray(saved) && saved.length < 1) ?
			(
				<BannerCard
					title={specialEvents.name}
					image={specialEvents.logo}
					onPress={() => Actions.SpecialEventsView({ personal: false })}
					onClose={() => hideCard('specialEvents')}
				/>
			) :
			(
				<Card
					title={specialEvents.name}
					id="specialEvents"
				>
					<View
						style={styles.contentContainer}
					>
						<SpecialEventsListView
							scrollEnabled={false}
							personal={true}
							rows={4}
							disabled={true}
							style={styles.specialEventsListView}
							inCard={true}
						/>
						<Touchable
							onPress={() => Actions.SpecialEventsView({ personal: true })}
						>
							<View style={styles.more}>
								<Text style={styles.more_label}>
									See My Schedule
								</Text>
							</View>
						</Touchable>
					</View>
				</Card>
			)
		}
	</View>
);

const styles = StyleSheet.create({
	more: { alignItems: 'center', justifyContent: 'center', padding: 6 },
	more_label: { fontSize: 20, color: COLOR_PRIMARY, fontWeight: '300' },
	specialEventsListView: { borderBottomWidth: 1, borderBottomColor: COLOR_MGREY },
	contentContainer: { flexGrow: 1, width: MAX_CARD_WIDTH },
});

export default SpecialEventsCard;
