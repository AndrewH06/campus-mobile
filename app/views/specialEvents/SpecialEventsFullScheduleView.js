import React from 'react';
import { View } from 'react-native';

import css from '../../styles/css';
import SpecialEventsListView from './SpecialEventsListView';

const SpecialEventsFullScheduleView = () => (
	<View
		style={[css.main_container, css.whitebg]}
	>
		<SpecialEventsListView
			scrollEnabled={true}
			personal={false}
		/>
	</View>
);

export default SpecialEventsFullScheduleView;
