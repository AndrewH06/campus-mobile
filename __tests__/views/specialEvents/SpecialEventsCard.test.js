import 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

// Component to be tested
import { SpecialEventsCard } from '../../../app/views/specialEvents/SpecialEventsCard';
import { SPECIAL_EVENTS_RESPONSE } from '../../../mockApis/specialEventsApi';

Enzyme.configure({ adapter: new Adapter() });

// Mock props passed down from state
const initialState = {
	specialEvents: SPECIAL_EVENTS_RESPONSE,
	saved: [],
	hideCard: jest.fn()
};

// Set up component to be rendered
function setup(props) {
	return shallow(<SpecialEventsCard {...props} />);
}

test('renders without crashing', () => {
	const tree = setup(initialState);
	expect(tree).toMatchSnapshot();
});
