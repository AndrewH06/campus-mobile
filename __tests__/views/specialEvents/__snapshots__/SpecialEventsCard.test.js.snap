// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View>
  <BannerCard
    image="https://s3-us-west-2.amazonaws.com/ucsd-its-wts/images/v1/campus_lisa_2017_logo.png"
    onClose={[Function]}
    onPress={[Function]}
    title="Campus Lisa"
  />
</View>
`;
